ui/selectors/selectors.js

```diff
@@ -1089,6 +1089,50 @@ export const getSnap = createDeepEqualSelector(
   },
 );
 
+/**
+ * Get a selector that returns the snap manifest for a given `snapId`.
+ *
+ * @param {object} state - The Redux state object.
+ * @param {string} snapId - The snap ID to get the manifest for.
+ * @returns {object | undefined} The snap manifest.
+ */
+export const getSnapManifest = createDeepEqualSelector(
+  (state) => state.metamask.currentLocale,
+  (state, snapId) => getSnap(state, snapId),
+  (locale, snap) => {
+    if (!snap?.localizationFiles) {
+      return snap?.manifest;
+    }
+
+    return getLocalizedSnapManifest(
+      snap.manifest,
+      locale,
+      snap.localizationFiles,
+    );
+  },
+);
+
+/**
+ * Get a selector that returns the snap metadata (name and description) for a
+ * given `snapId`.
+ *
+ * @param {object} state - The Redux state object.
+ * @param {string} snapId - The snap ID to get the metadata for.
+ * @returns {object} An object containing the snap name and description.
+ */
+export const getSnapMetadata = createDeepEqualSelector(
```

Comment:
Maybe we need a selector that contains all the metadata for all snaps? That would be easily passed down through the permission copy logic. We can keep one for accessing just the metadata of one snap though

Reply:
ok, will add.